src/common/dsp/effect/CombulatorEffect.cpp

```diff
@@ -0,0 +1,320 @@
+            {
+                Reg[e][c][q] = 0;
+            }
+        }
+    }
+}
+
+CombulatorEffect::~CombulatorEffect() { _aligned_free(qfus); }
+
+void CombulatorEffect::init()
+{
+    setvars(true);
+    bi = 0;
+}
+
+void CombulatorEffect::setvars(bool init)
+{
+    if (init)
+    {
+
+        for (int i = 0; i < 3; ++i)
+        {
+            cutoff[i].instantize();
+            resonance[i].instantize();
+            bandGain[i].instantize();
+        }
+
+        mix.set_target(1.f);
+
+        width.instantize();
+        mix.instantize();
+
+        halfbandOUT.reset();
+        halfbandIN.reset();
+    }
+    else
+    {
+        for (int i = 0; i < 3; ++i)
+        {
+            cutoff[i].newValue(*f[combulator_freq1 + i * 3]);
+            resonance[i].newValue(*f[combulator_feedback]);
+            bandGain[i].newValue(amp_to_linear(*f[combulator_gain1 + i * 3]));
+        }
+    }
+
+    width.set_target_smoothed(db_to_linear(*f[combulator_width]));
+}
+
+inline void set1f(__m128 &m, int i, float f)
+{
+    *((float *)&m + i) = f;
+}
+
+inline float get1f(__m128 m, int i)
+{
+    return *((float *)&m + i);
+}
+
+void CombulatorEffect::process(float *dataL, float *dataR)
+{
+    if (bi == 0)
+    {
+        setvars(false);
+    }
+    bi = (bi + 1) & slowrate_m1;
+
+    /*
+     * Strategy: Upsample, Set the coefficients, run the filter, Downsample
+     */
+
+    // Upsample phase. This wqorks and needs no more attention.
+    float dataOS alignas(16)[2][BLOCK_SIZE_OS];
+    halfbandIN.process_block_U2(dataL, dataR, dataOS[0], dataOS[1]);
+
+    /*
+     * Select the coefficients. Here you have to base yourself on the mode switch and
+     * call CoeffMaker with one fo the known models and subtypes, along with the resonance
+     * and the frequency of the particular band.
+     */
+    int type = fut_comb_pos, subtype = 1;
+
+    FilterUnitQFPtr filtptr = GetQFPtrFilterUnit(type, subtype);
+
+    for (int i = 0; i < 3; ++i)
+    {
+        cutoff[i].newValue(*f[combulator_freq1 + i * 3]);
+        resonance[i].newValue(*f[combulator_feedback]);
+        bandGain[i].newValue(amp_to_linear(*f[combulator_gain1 + i * 3]));
+    }
+
+    /*
+     * So now set up across the voices (e for 'entry' to match SurgeVoice) and the channels (c)
+     */
+    for (int e = 0; e < 3; ++e)
+    {
+        for (int c = 0; c < 2; ++c)
+        {
+            coeff[e][c].MakeCoeffs(cutoff[e].v, resonance[e].v, type, subtype, storage);
+
+            for (int i = 0; i < n_cm_coeffs; i++)
+            {
+                set1f(qfus[c].C[i], e, coeff[e][c].C[i]);
+                set1f(qfus[c].dC[i], e, coeff[e][c].dC[i]);
+            }
+
+            for (int i = 0; i < n_filter_registers; i++)
+            {
+                set1f(qfus[c].R[i], e, Reg[e][c][i]);
+            }
+
+            qfus[c].DB[e] = filterDelay[e][c];
+            qfus[c].WP[e] = WP[e][c];
+            qfus[c].WP[0] = subtype;
```

Comment:
This will be where the bug is by the way. For sure. This needs to be conditioned on filter type (or removed here)